src/components/Articles.jsx

```diff
@@ -1,31 +1,53 @@
-import { fetchAllArticles} from "../../api";
+import { fetchAllArticles,fetchAllArticlesByTopic} from "../../api";
 import {useState,useEffect} from 'react';
 import ArticlesCard from "./ArticlesCard"; 
 import styles from "../css/Articles.module.css"
+import { useSearchParams } from 'react-router-dom';
 
 
 
 
-function Articles() {
+function Articles({topicQuery}) {
 
       const [articles,setArticles] = useState([])
-      const [isLoading, setIsLoading] = useState(true)
+     
+
+      const [searchParams, setSearchParams] = useSearchParams();
+        
+      const sortByQuery = searchParams.get('topic');
+     
+     
+
 
    
     useEffect(()=>{
-      fetchAllArticles().then(({articles})=>{
-      setArticles(articles)
-       setIsLoading(false)
-      })
-    },[isLoading])
+      if(!sortByQuery) {
+        fetchAllArticles().then(({articles})=>{
+          setArticles(articles)
+        
+          })
+      }
+     
+    },[])
 
-   
     
-     if(isLoading) return <p>Loading....</p> 
+    useEffect(()=>{
+
+      if(sortByQuery) {
```

Comment:
You might want to change this variable name later, once we get to the sort by ticket (what we're currently doing is a filter)